apps/web/src/services/asset-inventory/components/AddCustomMetricModal.vue

```diff
@@ -48,38 +73,95 @@ const {
 
 
 /* Event */
-const handleClose = () => { state.proxyVisible = false; };
+const handleClose = () => {
+    resetForm();
+    state.proxyVisible = false;
+};
+
+const handleConfirm = async () => {
+    try {
+        const jsonParsedQuery = JSON.parse(code.value.trim());
+        const createParameters: MetricCreateParameters = {
+            name: name.value,
+            unit: unit.value,
+            metric_type: METRIC_TYPE.GAUGE,
+            resource_type: 'inventory.CloudService',
+            query_options: jsonParsedQuery,
+            namespace_id: metricExplorerPageStore.state.selectedNamespace?.namespace_id || '',
+        };
+        await SpaceConnector.clientV2.inventory.metric.create<MetricCreateParameters, MetricModel>(createParameters);
+        resetForm();
+        state.proxyVisible = false;
+        showSuccessMessage(i18n.t('INVENTORY.METRIC_EXPLORER.CUSTOM_METRIC.SUCCESSFULLY_ADDED'), '');
+    } catch (e) {
+        ErrorHandler.handleError(e);
+    }
+};
+
+/* Method */
+const resetForm = () => {
+    setForm('name', '');
+    setForm('unit', '');
+    setForm('code', '');
+};
```

Comment:
nitpick: there's `initForm()` in `useFormValidator` hook

Reply:
Oh thanks~!